tap_hubspot/__init__.py

```diff
@@ -1104,6 +1112,85 @@ def sync_deal_pipelines(STATE, ctx):
     singer.write_state(STATE)
     return STATE
 
+def gen_request_custom_objects(tap_stream_id, url, params, path, more_key):
+    """
+    Cursor-based API Pagination : Used in custom_objects stream implementation
+    """
+    with metrics.record_counter(tap_stream_id) as counter:
+        while True:
+            data = request(url, params).json()
+
+            if data.get(path) is None:
+                raise RuntimeError(
+                    "Unexpected API response: {} not in {}".format(path, data.keys()))
+
+            for row in data[path]:
+                counter.increment()
+                yield row
+
+            if not data.get(more_key):
+                break
+            params['after'] = data.get(more_key).get('next').get('after')
```

Comment:
Chains of `.get()` calls are not as safe as you think.

Consider the following scenarios. Which ones print and which ones error?

A.
```python
data = {}
more_key = "paging"

print(data.get(more_key).get('next').get('after'))
```

B.
```python
data = {"paging": {}}
more_key = "paging"

print(data.get(more_key).get('next').get('after'))
```

C.
```python
data = {"paging": {"next": {}}}
more_key = "paging"

print(data.get(more_key).get('next').get('after'))
```

Reply:
[spoilers.txt](https://github.com/singer-io/tap-hubspot/files/13337552/spoilers.txt)

Reply:
[spoilers](https://gist.github.com/luandy64/e0ba5d71ab6415190ff572f1c63e8369)

Reply:
Attribute error for options A and B. Option C gives us the default value as None.

In this case, I think I can modify as below - 
```
params['after'] = data.get(more_key, {}).get('next', {}).get('after', None)
if params['after'] is None:
    break
```